include/engine/search_engine_data.hpp

```diff
@@ -37,13 +53,23 @@ struct SearchEngineData
 
     using ManyToManyHeapPtr = boost::thread_specific_ptr<ManyToManyQueryHeap>;
 
+    using MultiLayerDijkstraHeap = util::BinaryHeap<NodeID,
+                                                    NodeID,
+                                                    EdgeWeight,
+                                                    MultiLayerDijkstraHeapData,
+                                                    util::UnorderedMapStorage<NodeID, int>>;
+
+    using MultiLayerDijkstraHeapPtr = boost::thread_specific_ptr<MultiLayerDijkstraHeap>;
+
     static SearchEngineHeapPtr forward_heap_1;
     static SearchEngineHeapPtr reverse_heap_1;
     static SearchEngineHeapPtr forward_heap_2;
     static SearchEngineHeapPtr reverse_heap_2;
     static SearchEngineHeapPtr forward_heap_3;
     static SearchEngineHeapPtr reverse_heap_3;
     static ManyToManyHeapPtr many_to_many_heap;
+    static MultiLayerDijkstraHeapPtr mld_forward_heap;
```

Comment:
yes, `HeapData::from_clique_arc` is only needed, but it will 4 bytes with alignment, so i placed complete edge ID. In CH the same flag is hidden in `QueryEdge` but i don't think it is good idea to cut one bit from node ID in `EdgeBasedEdge``